### ISSUE TITLE
Inspect hangs when the user creates a circular dependency for the ListViewGroup and ListViewItem

### ISSUE THREAD

**Issue by SergeySmirnov-Akvelon** [dotnet/winforms#4045]
.NET Core Version:
.NET SDK 5.0.100-rc.1.20420.14

Have you experienced this same bug with .NET Framework?:
No

**Minimal repro:**
1. Create application with list view.
2. Create ListViewGroup
3. Create ListViewItem linked to ListViewGroup
4. Add ListViewItem to ListViewGroup
5. Add ListViewGroup to ListView
6. Add ListViewItem to ListView
7. Run application.
8. Run Inspector.
9. Try to get data about ListViewItem in Inspector.
10. Or just user attached [WindowsFormsApp3.zip](https://github.com/dotnet/winforms/files/5310765/WindowsFormsApp3.zip)

**Expected behavior:**
Data about selected ListViewItem should be displayed.

**Actual behavior:**
Inspector hangs or return that the ListViewGroup contains a huge number of ListViewItems
![ListViewAndInspector](https://user-images.githubusercontent.com/23376742/94789698-edf76000-03dd-11eb-9c85-348bad3717f3.jpg)

**Comment by Tanya-Solyanik** [comment]
We'll consider this case a user error
